### ISSUE TITLE
Question about Reproducing Results

### ISSUE THREAD

**Issue by ubless607** [woyut/WSAG-PLSP#3]
Hello, @woyut.
Thank you for sharing your impressive work.

I have a question regarding the training code provided in the `README.md`. Specifically, does the following command matches the _Ours-full_ result reported in the paper?
```
python train.py --config configs/seen.yaml --seed 10000
```

I attempted to reproduce the results using this command, but obtained the following metrics:
```
KLD=0.951, SIM=0.517, NSS=1.53
```
While the SIM and NSS scores seem reasonable, the KLD is significantly higher than expected. 
Can you suggest any potential issues that might be affecting the KLD?

Thank you in advance for your help.

p.s What is the score after each training epoch? Is this the validation score?

**Comment by woyut** [comment]
Hello! Thanks for your interest in this work!
The command you mentioned should match our "full" model. However, the final results are subject to a certain degree of variability due to randomness. Therefore, the reported performance is obtained from the average of five independent training runs. You may consider trying different random seeds, which could potentially lead to better KLD. 

In addition, the scores shown after each epoch correspond to the test set, as AGD20K does not provide a validation split.

Below is the training log from one of our runs, for your reference.
============Training Epoch 0============
Training loss: 20.284559859621055, KL loss: 1.4803850903040268, Sim loss: 0.13630596586389093, Exo CLS loss: 1.7163170133459256, 
Noun sim loss: 0.2028588334799639, Part sim loss: 0.7508617231394347
learning rate:0.0001

Result on AGD: 
KLD=1.1096751689910889, SIM=0.4262408912181854, NSS=1.2856957912445068
noun sim: 0.11825910575136649, part sim: 0.4900311199605892
New best KLD: 1.1096751689910889, 0.4262408912181854, 1.2856957912445068
============Training Epoch 1============
Training loss: 11.603591313951286, KL loss: 1.0370221946551947, Sim loss: 0.06626407795974998, Exo CLS loss: 0.9740002270789366, 
Noun sim loss: 0.12484087364416888, Part sim loss: 0.3908519415189157
learning rate:0.0001

Result on AGD: 
KLD=1.091105580329895, SIM=0.418358713388443, NSS=1.4187275171279907
noun sim: 0.11186951565998551, part sim: 0.3226222152641679
New best KLD: 1.091105580329895, 0.418358713388443, 1.4187275171279907
============Training Epoch 2============
Training loss: 9.51752641845245, KL loss: 0.9325846165384211, Sim loss: 0.058209082308520003, Exo CLS loss: 0.7852235612787218, 
Noun sim loss: 0.12043169580881782, Part sim loss: 0.30183675911010827
learning rate:0.0001

Result on AGD: 
KLD=1.0025664567947388, SIM=0.46090561151504517, NSS=1.410841703414917
noun sim: 0.10925724549410167, part sim: 0.249285633033386
New best KLD: 1.0025664567947388, 0.46090561151504517, 1.410841703414917
============Training Epoch 3============
Training loss: 8.228225696513615, KL loss: 0.8311644794578571, Sim loss: 0.056243801275461774, Exo CLS loss: 0.6688320397156352, 
Noun sim loss: 0.12034047983036168, Part sim loss: 0.25962335548382987
learning rate:0.0001

Result on AGD: 
KLD=0.9744734764099121, SIM=0.4737194776535034, NSS=1.4600818157196045
noun sim: 0.10629101542129152, part sim: 0.24218604082423917
New best KLD: 0.9744734764099121, 0.4737194776535034, 1.4600818157196045
============Training Epoch 4============
Training loss: 7.595611958560162, KL loss: 0.7642534107408608, Sim loss: 0.05032718803101234, Exo CLS loss: 0.6182555731200233, 
Noun sim loss: 0.1215097487755722, Part sim loss: 0.24021222159335298
learning rate:0.0001

Result on AGD: 
KLD=0.966657817363739, SIM=0.4657295346260071, NSS=1.5200276374816895
noun sim: 0.10375930749431146, part sim: 0.2268909090031872
New best KLD: 0.966657817363739, 0.4657295346260071, 1.5200276374816895
============Training Epoch 5============
Training loss: 7.032692018509186, KL loss: 0.7152191582404713, Sim loss: 0.051838399596107955, Exo CLS loss: 0.5654770485382755, 
Noun sim loss: 0.12047810908449991, Part sim loss: 0.23840268721102842
learning rate:0.0001

Result on AGD: 
KLD=0.916204035282135, SIM=0.49004843831062317, NSS=1.5194305181503296
noun sim: 0.1008049966167294, part sim: 0.22924549290843682
New best KLD: 0.916204035282135, 0.49004843831062317, 1.5194305181503296
============Training Epoch 6============
Training loss: 6.777186249535505, KL loss: 0.6737649510324207, Sim loss: 0.04958608610164073, Exo CLS loss: 0.5464982295746605, 
Noun sim loss: 0.11942419666134457, Part sim loss: 0.23153953859905171
learning rate:0.0001

Result on AGD: 
KLD=0.8980460166931152, SIM=0.5065261125564575, NSS=1.5325942039489746
noun sim: 0.09752482945779628, part sim: 0.22087296615921512
New best KLD: 0.8980460166931152, 0.5065261125564575, 1.5325942039489746
============Training Epoch 7============
Training loss: 6.449699041095947, KL loss: 0.6466797201122064, Sim loss: 0.04908063009329674, Exo CLS loss: 0.5171138141054217, 
Noun sim loss: 0.11853736756459808, Part sim loss: 0.22537508574217627
learning rate:0.0001

Result on AGD: 
KLD=0.8811210989952087, SIM=0.5056421756744385, NSS=1.5739320516586304
noun sim: 0.09482963001777847, part sim: 0.2183938697869567
New best KLD: 0.8811210989952087, 0.5056421756744385, 1.5739320516586304
============Training Epoch 8============
Training loss: 6.316486713861483, KL loss: 0.6201811485319435, Sim loss: 0.05203578706645766, Exo CLS loss: 0.5035349689444832, 
Noun sim loss: 0.11810576952305595, Part sim loss: 0.22492232977242724
learning rate:0.0001

Result on AGD: 
KLD=0.8899584412574768, SIM=0.5087466835975647, NSS=1.5523937940597534
noun sim: 0.094629480287895, part sim: 0.21247590302994257
============Training Epoch 9============
Training loss: 6.225244334264385, KL loss: 0.6017272403425006, Sim loss: 0.05416137565951149, Exo CLS loss: 0.4942331841529482, 
Noun sim loss: 0.11765229299354774, Part sim loss: 0.21919206214109413
learning rate:0.0001

Result on AGD: 
KLD=0.9079084992408752, SIM=0.5006369948387146, NSS=1.5642189979553223
noun sim: 0.09465737640857697, part sim: 0.21112468892037156
============Training Epoch 10============
Training loss: 5.9192246654980725, KL loss: 0.5738712631350955, Sim loss: 0.049197941331308996, Exo CLS loss: 0.47151579093806645, 
Noun sim loss: 0.11676674021739195, Part sim loss: 0.21449336125655208
learning rate:0.0001

Result on AGD: 
KLD=0.8716228604316711, SIM=0.5154688358306885, NSS=1.569014072418213
noun sim: 0.09444124387634967, part sim: 0.20466638482698107
New best KLD: 0.8716228604316711, 0.5154688358306885, 1.569014072418213
============Training Epoch 11============
Training loss: 5.925198532640934, KL loss: 0.548997966006966, Sim loss: 0.05172604376420069, Exo CLS loss: 0.4721336778413387, 
Noun sim loss: 0.1163113235472325, Part sim loss: 0.21292026460549296
learning rate:0.0001

Result on AGD: 
KLD=0.9113656878471375, SIM=0.5086357593536377, NSS=1.5679113864898682
noun sim: 0.09024360459431259, part sim: 0.19636939931883163
============Training Epoch 12============
Training loss: 5.787905213584388, KL loss: 0.5443309940630032, Sim loss: 0.05035801751401456, Exo CLS loss: 0.4603422843903862, 
Noun sim loss: 0.11604193727513797, Part sim loss: 0.2052925676193441
learning rate:0.0001

Result on AGD: 
KLD=0.9353938698768616, SIM=0.5168824791908264, NSS=1.5076922178268433
noun sim: 0.08897092039552396, part sim: 0.19831181495385977
============Training Epoch 13============
Training loss: 5.603152114579639, KL loss: 0.5204398656741438, Sim loss: 0.050151557931525946, Exo CLS loss: 0.44454641170530296, 
Noun sim loss: 0.11548327966474908, Part sim loss: 0.20249269420913404
learning rate:0.0001

Result on AGD: 
KLD=0.8993480205535889, SIM=0.5129567980766296, NSS=1.5677098035812378
noun sim: 0.08799000070174724, part sim: 0.19003630587764458
============Training Epoch 14============
Training loss: 5.69988953146662, KL loss: 0.5006909304433965, Sim loss: 0.05048635212261462, Exo CLS loss: 0.45596089740148277, 
Noun sim loss: 0.11513826694107221, Part sim loss: 0.19587821876479627
learning rate:0.0001

Result on AGD: 
KLD=0.927262544631958, SIM=0.5092713236808777, NSS=1.5324523448944092
noun sim: 0.08830626003959867, part sim: 0.18632984726838678
============Training Epoch 15============
Training loss: 5.477319689153524, KL loss: 0.4820530566518081, Sim loss: 0.048085825971605184, Exo CLS loss: 0.4379939034589406, 
Noun sim loss: 0.11489851849355819, Part sim loss: 0.19570801143833452
learning rate:0.0001

Result on AGD: 
KLD=0.9332786798477173, SIM=0.5165948271751404, NSS=1.4927825927734375
noun sim: 0.08692513809923909, part sim: 0.18653951537281915
============Training Epoch 16============
Training loss: 5.465747376185504, KL loss: 0.4688818627284457, Sim loss: 0.050273992982639244, Exo CLS loss: 0.43598148104817486, 
Noun sim loss: 0.11493726776306266, Part sim loss: 0.19373513649348986
learning rate:0.0001

Result on AGD: 
KLD=0.9203565120697021, SIM=0.5126316547393799, NSS=1.534787893295288
noun sim: 0.08552723318373957, part sim: 0.18370105857851965
============Training Epoch 17============
Training loss: 5.460598416239298, KL loss: 0.4539241315904514, Sim loss: 0.05225865786473587, Exo CLS loss: 0.4350093730713259, 
Noun sim loss: 0.11471426174233884, Part sim loss: 0.1927970849605617
learning rate:0.0001

Result on AGD: 
KLD=0.9193922281265259, SIM=0.5276476144790649, NSS=1.5502232313156128
noun sim: 0.0875634076007084, part sim: 0.18106508084299455
============Training Epoch 18============
Training loss: 5.421163087697833, KL loss: 0.44132477623640665, Sim loss: 0.05212906870978087, Exo CLS loss: 0.43253356561729533, 
Noun sim loss: 0.11439368646273575, Part sim loss: 0.18818303319791171
learning rate:0.0001

Result on AGD: 
KLD=0.9040764570236206, SIM=0.5245298743247986, NSS=1.5574814081192017
noun sim: 0.08619157805469839, part sim: 0.1811367410250666
============Training Epoch 19============
Training loss: 5.190751382715432, KL loss: 0.4268730220322521, Sim loss: 0.0514340936733242, Exo CLS loss: 0.41161755434373576, 
Noun sim loss: 0.11455321491332444, Part sim loss: 0.18808674638957543
learning rate:0.0001

Result on AGD: 
KLD=0.9215685129165649, SIM=0.5208470225334167, NSS=1.5527338981628418
noun sim: 0.08599791497098802, part sim: 0.1752705876326504
